src/ragger/backend/speculos.py

```diff
@@ -241,6 +241,8 @@ def get_current_screen_content(self) -> dict:
         return self._retrieve_client_screen_content()
 
     def compare_screen_with_text(self, text: str) -> bool:
+        # Update self._last_screenshot to keep coherent internal state
+        self._last_screenshot = BytesIO(self._client.get_screenshot())
         for event in self._retrieve_client_screen_content()["events"]:
             if match(text, event.get("text", "")):
                 return True
```

Comment:
Shouldn't it be just before the return?

Reply:
Not sure it changes anything, the `for` loop is only a few elements long
We want to be the closest to `_retrieve_client_screen_content`
I've considered putting it inside `_retrieve_client_screen_content` but I'm worried of side effects

Reply:
you could retrieved the events, then update the screenshot, and eventually loop on the event. WDYT?